internal/provider/resource_self_managed_worker.go

```diff
@@ -0,0 +1,287 @@
+	address                            = "address"
+	canonicalTags                      = "canonical_tags"
+	configTags                         = "config_tags"
+	lastStatusTime                     = "last_status_time"
+	workerGeneratedAuthToken           = "worker_generated_auth_token"
+	controllerGeneratedActivationToken = "controller_generated_activation_token"
+	apiTags                            = "api_tags"
+	releaseVersion                     = "release_version"
+	authorizedActions                  = "authorized_actions"
+)
+
+func resourceSelfManagedWorker() *schema.Resource {
+	return &schema.Resource{
+		Description: "The resource allows you to create a self-managed worker object.",
+
+		CreateContext: resourceSelfManagedWorkerCreate,
+		ReadContext:   resourceSelfManagedWorkerRead,
+		UpdateContext: resourceSelfManagedWorkerUpdate,
+		DeleteContext: resourceSelfManagedWorkerDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			IDKey: {
+				Description: "The ID of the worker.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			ScopeIdKey: {
+				Description: "The scope for the worker.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			NameKey: {
+				Description: "The name for the worker.",
+				Type:        schema.TypeString,
+				Computed:    false,
+				Optional:    true,
+			},
+			DescriptionKey: {
+				Description: "The description for the worker.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			createdTime: {
+				Description: "When the self managed worker was created.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			version: {
+				Description: "The internal version of the self managed worker resource.",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+			address: {
+				Description: "The accessible address of the self managed worker.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			canonicalTags: {
+				Description: "",
+				Type:        schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeMap,
+				},
+				Computed: true,
+			},
+			configTags: {
+				Description: "",
+				Type:        schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeMap,
+				},
+				Computed: true,
+			},
+			workerGeneratedAuthToken: {
+				Description: "The worker authentication token required to register the worker for the worker-led authentication flow. Leaving this blank will result in a controller generated token.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			controllerGeneratedActivationToken: {
+				Description: "A single use token generated by the controller to be passed to the self-managed worker.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			TypeKey: {
+				Description: "The type of self managed worker. Will be one of PKI or KMS.",
```

Comment:
We normally separate resources based on `type` instead of having this provided as a schema, for example we have a vault and static resource for credential stores, doing this allows us to remove complexity in optional schemas. Here it would make since `KMS` workers cannot be created or even updated via the TF the best we can do is set worker tags.